crypto/x509/x_attrib.c

```diff
@@ -56,3 +57,228 @@ X509_ATTRIBUTE *X509_ATTRIBUTE_create(int nid, int atrtype, void *value)
     ASN1_TYPE_free(val);
     return NULL;
 }
+
+static int print_hex(BIO *out, unsigned char *buf, int len)
+{
+    int i;
+
+    for (i = 0; i < len; i++) {
+        if (BIO_printf(out, "%02X ", buf[i]) <= 0) {
+            return 0;
+        }
+    }
+    return 1;
+}
+
+static int asn1_integer_print_bio(BIO *bio, const ASN1_INTEGER *num)
+{
+    BIGNUM *num_bn;
+    int result = 0;
+    char *hex;
+
+    num_bn = ASN1_INTEGER_to_BN(num, NULL);
+    if (num_bn == NULL)
+        return -1;
+    if ((hex = BN_bn2hex(num_bn)) != NULL) {
+        result = BIO_write(bio, "0x", 2) > 0;
+        result = result && BIO_write(bio, hex, strlen(hex)) > 0;
+        OPENSSL_free(hex);
+    } else {
```

Comment:
set `result = 0;` just after `OPENSSL_free()`.
So you can remove the `else` block.